### PR TITLE
[AND-4953] adapter changes to support in-feed videos

### PR DESCRIPTION
**Pull request by HuFangshuai**
pr for ticket: https://vungle.atlassian.net/browse/AND-4953

### CODE REVIEW

File: Adapters/Vungle/ISVungleAdapter/src/main/java/com/ironsource/adapters/vungle/banner/VungleBannerAdapter.kt [hunk 1]
```diff
@@ -200,19 +200,43 @@ class VungleBannerAdapter(adapter: VungleAdapter) :
         }
     }
 
-    private fun getBannerSize(bannerSize: ISBannerSize): BannerAdSize? {
-        return when (bannerSize.description) {
-            "BANNER", "LARGE" -> BannerAdSize.BANNER
-            "RECTANGLE" -> BannerAdSize.VUNGLE_MREC
+    private fun getBannerSize(bannerSize: ISBannerSize): VungleAdSize? {
+        val vungleAdSize = when (bannerSize.description) {
```
Comment (ashinagawa) [inline Adapters/Vungle/ISVungleAdapter/src/main/java/com/ironsource/adapters/vungle/banner/VungleBannerAdapter.kt]: So, you set `vungleAdSize` based on `bannerSize.description` first.
Then, update `vungleAdSize` with `getAdSizeWithWidth` only when `isAdaptive` is `true`.

This logic could cover all cases, so I will follow this logic in iOS side as well.

File: Adapters/Vungle/ISVungleAdapter/src/main/java/com/ironsource/adapters/vungle/banner/VungleBannerAdapter.kt [hunk 2]
```diff
@@ -200,19 +200,43 @@ class VungleBannerAdapter(adapter: VungleAdapter) :
...
+            "CUSTOM" -> VungleAdSize.getAdSizeWithWidthAndHeight(
+                bannerSize.width,
+                bannerSize.height
+            )
+
             else -> null
         }
+
+        try {
+            if (bannerSize.isAdaptive && vungleAdSize != null) {
```
Comment (ashinagawa) [inline Adapters/Vungle/ISVungleAdapter/src/main/java/com/ironsource/adapters/vungle/banner/VungleBannerAdapter.kt:223]: From IS's integration guide, they recommend to set in their integration guide.
```
val bannerSize = ISBannerSize.BANNER// the banner size you choose will be the default value for non supporting banner adaptive networks
```
So, there should be a case in which `isAdaptive == true` and `vungleAdSize = VungleAdSize.BANNER` at this line?






https://developers.is.com/ironsource-mobile/android/banner-integration-android/#step-6

File: Adapters/Vungle/ISVungleAdapter/src/main/java/com/ironsource/adapters/vungle/banner/VungleBannerAdapter.kt (hunk 1 above)
Comment (ashinagawa) [reply Adapters/Vungle/ISVungleAdapter/src/main/java/com/ironsource/adapters/vungle/banner/VungleBannerAdapter.kt]: You will support "custom" size here, right?

Comment (HuFangshuai) [reply Adapters/Vungle/ISVungleAdapter/src/main/java/com/ironsource/adapters/vungle/banner/VungleBannerAdapter.kt]: Yes

**Review by stanley-vungle — APPROVED** [review]

**Review by ManojBudumuru-vungle — APPROVED** [review]

**Pull request merged**